### ISSUE TITLE
Mini-map covers the Puzzle map in Hide Interface mode

### ISSUE THREAD

**Issue by Districh-ru** [ihhub/fheroes2#6548]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

In Hide Interface mode the mini-map (with Exit button) stays in front of the Puzzle map on the place it was before opening, it covers the puzzle map and cannot be moved away.
![puzzle](https://user-images.githubusercontent.com/113276641/216619534-54b15a74-5ee6-4dec-9a54-c1a53587fb32.png)

### Save file

No save is needed.
1. Load any map or save.
2. In game settings set Interface -> Hide.
3. Place the mini-map in the center part of the screen.
4. Open puzzle map.

### Additional info

For 640x480 Puzzle map could be rendered like in Show Interface mode.
For higher resolutions the issue might be fixed by letting to move mini-map, or by forcing its position outside the Puzzle map.

**Comment by Districh-ru** [comment]
The same is for Teleport Spell :(
![изображение](https://user-images.githubusercontent.com/113276641/216630185-b46254b9-f79f-4f2e-ab7d-2b79e69a1337.png)

**Comment by oleg-derevenetz** [comment]
This is not quite a "bug", this is the consequence of "no interface" mode's feature which is that any window can be placed anywhere on the screen and cover any part of the map. Of course, it would be nice to be able to move an appropriate window in such "dialog mode" but essentially a user is perfectly able to move this window in advance (say, just don't place the radar at the center of the screen) or he can exit the "dialog mode", move the window and then redo the action, because no resources (such as mana) will be wasted if action will be canceled.

**Comment by Districh-ru** [comment]
Yes, the user can exit the Puzzle map or cancel the Dimension Door spell, but in resolutions, lower than 800x600 there is no room to place the mini-map so it won't cover the Puzzle map or the teleport area.

**Comment by oleg-derevenetz** [comment]
> Yes, the user can exit the Puzzle map or cancel the Dimension Door spell, but in resolutions, lower than 800x600 there is no room to place the mini-map so that it won't cover the Puzzle map or the teleport area.

"No interface" mode is in principle not very convenient at lower resolutions, because these additional windows are always overlap something on adventure map. It's just not meant for low res.

**Comment by Districh-ru** [comment]
So lets make it an improvement request for the ability to move mini-map in "dialog mode".
